tests/shakedown/shakedown/clients/authentication.py

```diff
@@ -0,0 +1,115 @@
+import json
+import logging
+import requests
+import toml
+
+from functools import lru_cache
+from os import environ, path
+from . import dcos_url, gen_url
+from ..errors import DCOSAuthenticationException
+
+
+logger = logging.getLogger(__name__)
+
+
+@lru_cache()
+def read_config():
+    """ Read configuration options from ~/.shakedown (if exists)
+        :return: a dict of arguments
+        :rtype: dict
+    """
+    configfile = path.expanduser('~/.shakedown')
+    args = dict()
+    if path.isfile(configfile):
+        with open(configfile, 'r') as f:
+            config = toml.loads(f.read())
+        for key in config:
+            param = key.replace('-', '_')
+            args[param] = config[key]
+    return args
+
+
+def authenticate(username, password):
+    """Authenticate with a DC/OS cluster and return an ACS token.
+    return: ACS token
+    """
+    url = gen_url('acs/api/v1/auth/login')
+
+    creds = {
+        'uid': username,
+        'password': password
+    }
+
+    response = requests.post(url, json=creds, headers={'Accept': 'application/json'})
+
+    response.raise_for_status()
+    return response.json()['token']
+
+
+def authenticate_oauth(oauth_token):
+    """Authenticate by checking for a valid OAuth token.
+    return: ACS token
+    """
+    url = gen_url('acs/api/v1/auth/login')
+    payload = {'token': oauth_token}
+    response = requests.post(url, json=payload, headers={'Content-Type': 'application/json'}, auth=None)
+
+    response.raise_for_status()
+    return response.json()['token']
+
+
+@lru_cache(1)
+def dcos_acs_token():
+    """Return the DC/OS ACS token as configured in the DC/OS library.
+    :return: DC/OS ACS token as a string
+    """
+    logger.info('Authenticating with DC/OS cluster...')
+
+    # Try token from dcos cli session
+   # try:
+   #     clusters = run_dcos_command('cluster list --attached --json')
```

Comment:
Importing `run_dcos_command` introduces a circular dependency since `shakedown.dcos` import `http` and `http` import `dcos_acs_token`. This will be resolved by [MARATHON-8415](https://jira.mesosphere.com/browse/MARATHON-8415/).